src/main/java/mnsky/TaskList.java

```diff
@@ -0,0 +1,255 @@
+     * @param stringIndex The index of the task to unmark in string format.
+     * @return The task that was unmarked.
+     * @throws MnskyInvalidParameterException If stringToIndex throws the exception.
+     */
+    public Task unmark(String stringIndex) throws MnskyInvalidParameterException {
+        int index = stringToIndex("unmark", stringIndex);
+        tasks.get(index).unmark();
+        return tasks.get(index);
+    }
+
+    /**
+     * Deletes the task corresponding to the given index parameter, if it exists.
+     * @param stringIndex The index of the task to delete in string format.
+     * @return The task that was deleted.
+     * @throws MnskyInvalidParameterException If stringToIndex throws the exception.
+     */
+    public Task delete(String stringIndex) throws MnskyInvalidParameterException  {
+        int index = stringToIndex("delete", stringIndex);
+        Task deletedTask = tasks.get(index);
+        tasks.remove(index);
+        return deletedTask;
+    }
+
+    /**
+     * Creates a LocalDate object based on the input string.
+     * @param input The input string to create a LocalDate object from
+     * @return The created LocalDate object if the input string is in a valid date format, null otherwise.
+     */
+    private LocalDate parseDate(String input) {
+        try {
+            return LocalDate.parse(input);
+        } catch (DateTimeParseException e) {
+            return null;
+        }
+    }
+
+    /**
+     * Creates a LocalTime object based on the input string.
+     * @param input The input string to create a LocalTime object from
+     * @return The created LocalTime object if the input string is in a valid time format, null otherwise.
+     */
+    private LocalTime parseTime(String input) {
+        try {
+            return LocalTime.parse(input);
+        } catch (DateTimeParseException e) {
+            return null;
+        }
+    }
+
+    /**
+     * Adds all the task in the storage data to the task list.
+     * @param storage The storage object.
+     * @throws MnskyException If there are any issues with converting the parsed storage data into tasks.
+     */
+    private void getStorageData(Storage storage) throws MnskyException {
+        ArrayList<ArrayList<String>> taskList = Parser.parseStorageData(storage.readFromDataFile());
+        for (ArrayList<String> task : taskList) {
+            Task actualTask;
+            switch (task.get(0)) {
+                case "task":
+                    actualTask = addTask(task.get(1));
+                    break;
+
+                case "event":
+                    actualTask = addEvent(task.get(1), task.get(2));
+                    break;
+
+                case "deadline":
+                    actualTask = addDeadline(task.get(1), task.get(2));
+                    break;
+
+                default:
+                    throw new MnskyException("????");
```

Comment:
Perhaps the Exception message could have more details for easier debugging?

Reply:
This exception is intended to never be thrown but I think I could make the message more helpful